src/Lib.hs

```diff
@@ -140,10 +168,11 @@ postBlacklistPool user blacklistPool = convertIOToHandler $ do
     return examplePoolOfflineMetadata
 
 -- throwError err404
-getPoolOfflineMetadata :: PoolHash -> Handler PoolOfflineMetadata
-getPoolOfflineMetadata poolHash = convertIOToHandler $ do
+getPoolOfflineMetadata :: DataLayer -> PoolHash -> Handler PoolMetadataWrapped
+getPoolOfflineMetadata dataLayer poolHash = convertIOToHandler $ do
     putTextLn $ show poolHash
-    return examplePoolOfflineMetadata
+    fmap PoolMetadataWrapped $ either (\m -> panic $ renderLookupFail m) (\a -> a) <$> (dlGetPoolMetadataSimple dataLayer) poolHash
```

Comment:
As a result, this returns a _string_ which encodes a JSON object, and not a JSON object.

Reply:
Yes. Encoding it into a JSON object can change the JSON contents, and accordingly, the hash.